openrouteservice/src/main/java/org/heigit/ors/routing/graphhopper/extensions/weighting/AdditionWeighting.java

```diff
@@ -20,146 +20,21 @@
 
 public class AdditionWeighting extends AbstractWeighting {
 	private Weighting superWeighting;
-    private WeightCalc weightCalc;
+	private Weighting[] weightings;
 
     public AdditionWeighting(Weighting[] weightings, Weighting superWeighting, FlagEncoder encoder) {
         super(encoder);
         this.superWeighting = superWeighting;
-        
-        int count = weightings.length;
-        if (count == 1)
-           weightCalc = new OneWeightCalc(weightings);
-        else if (count == 2)
-            weightCalc = new TwoWeightCalc(weightings);
-        else if (count == 3)
-            weightCalc = new ThreeWeightCalc(weightings);
-        else if (count == 4)
-            weightCalc = new FourWeightCalc(weightings);
-        else if (count == 5)
-            weightCalc = new FiveWeightCalc(weightings);
+        this.weightings = weightings.clone();
```

Comment:
It is necessary to preserve the behavior. Otherwise, if the call-site would change the array after construction of the addition weighting, the latter would change under the hood, which was not the case in the previous implementation.

Reply:
True. But do we actually need an _array_ in first place? Originally the weightings are bundled into a `List`, so maybe just skip the format conversion altogether and just pass the list. The weightings list in `ORSWeightingFactory.createWeighting()` is a local variable created just for constructing `AdditionWeighting` and not referenced elsewhere, so even though I agree about the potential threat of post-modification, I don't think this is of an issue here. What do you think?

Reply:
In principle I agree, but I would prefer to do this in a second step because it has a broader scope and a bigger impact. My intention with this PR was to fix the bugs and clean up the biggest mess in a small, self-contained PR. All changes are local to `AdditionWeighting` and keep the interfaces stable. For changing interfaces we need more tests to refactor safely. IMO the whole model around the soft weightings looks unsound and would need to be redone anyways.

Reply:
Sure, it's ultimately your call. I was suggesting this because `AdditionWeighting` is called only by `createWeighting` method so adjusting the interface didn't sound like a big deal to me, still a local change in a sense. While the mathematical model behind the soft weighting needs revising, the suggested modification is just a change to the implementation. Personally I would have done it in this PR in order to get this off the table and focus on the logic in any follow up PRs.